### DISCUSSION TITLE
Optional Properties?

### DISCUSSION THREAD

**Discussion by okybr** [oapi-codegen/oapi-codegen#652] [Q&A] [answered]
Hello :)

Sometimes an object within an API-specification has optional properties. Suppose the API manages a database with a table "user" with columns / properties "name" and "nick". The API could serve a route to update a user. Both the name- and the age-property would be optional in the payload to this update-route.

**Is `oapi-codegen` able to generate code with optional properties?**

In Go, you can use a struct as an "option type":

```go
type Maybe[T any] struct {
	state bool
	value T
}
```

**Is `oapi-codegen` able to generate code which represents optional properties like this?**

**Comment by jamietanna** [comment, marked as answer]
Hey, we currently do this using pointers to denote the value may not be there, i.e.:

https://github.com/deepmap/oapi-codegen/blob/9a9e66fe11c82b21732777233127d518ad092481/examples/petstore-expanded/gorilla/api/petstore.gen.go#L30-L37

We're not yet targeting Go 1.18, so won't be able to use generics/`any` just yet

**Comment by kotyara85** [comment]
@jamietanna you have any plans to implement this yet?
